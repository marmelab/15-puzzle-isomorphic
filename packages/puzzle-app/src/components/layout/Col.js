import React from 'react';
import PropTypes from 'prop-types';

const Col = ({ children }) => <div className="col s12">{children}</div>;

Col.propTypes = {
    children: PropTypes.oneOfType([
        PropTypes.element,
        PropTypes.arrayOf(PropTypes.element),
    ]).isRequired,
};

export default Col;
