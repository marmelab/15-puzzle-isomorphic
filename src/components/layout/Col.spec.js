import React from 'react';
import renderer from 'react-test-renderer';

import Col from './Col';

describe('Col', () => {
    test('should render Col without error', () => {
        renderer.create(<Col />);
    });

    test('should render the Col correctly', () => {
        const component = renderer.create(<Col />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
