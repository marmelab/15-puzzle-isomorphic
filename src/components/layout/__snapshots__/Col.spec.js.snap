// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Col should render the Col correctly 1`] = `
<div
  className="col s12"
>
  <p>
    Child
  </p>
</div>
`;
